### ISSUE TITLE
Umbraco is not validating same way save user and change password

### ISSUE THREAD

**Issue by bielu** [umbraco/Umbraco-CMS#14823]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.6.1

### Bug summary

Bug found as part: https://github.com/umbraco/Umbraco-CMS/issues/14808

### Steps to reproduce

1. set UsernameIsEmail to false
2. create user 
3. rename user with name containing a space in login
4. try to change password / unlock / disable user
5. you will get notication about username is incorrect

### Expected result / actual result

validation is same between save user and change password

**Comment by andr317c** [comment]
Hey!
Thanks a lot for reporting this issue, I was able to reproduce the issue on versions 10.6.1, 10.7.0, and 12.2.0. 

What I did was: Set `UsernameIsEmail` to false in the app settings, and create a user. Change the Username to contain a space and save the user. If I then tried to change the password of the user, I ran into the validation error: `Validation: Invalid username`. I was able to Disable and Enable the user without any issues. I was also able to log in to the user even though its username contained a space. 

I will mark this issue up as an up for grabs! 😄 Please let me know if I missed anything.

**Comment by Amalie-Wowern** [comment]
I will try to look at this

**Comment by bielu** [comment]
@Amalie-Wowern it is ltitle tricky as Umbraco use .net identity abstraction to validate and when you saving you have Ipublishcontent. :) If you need support about code logic there let me know

**Comment by Amalie-Wowern** [comment]
@bielu Yes i noticed its a little tricky.
I would love some help about the code logic

**Comment by Amalie-Wowern** [comment]
If i understand the issue correctly.

The bug is that, when the password is changed its validating the username wrong. It does not take into consideration that the UsernameIsEmail is disabled?

**Comment by bielu** [comment]
> The bug is that, when the password is changed its validating the username wrong. It does not take into consideration that the UsernameIsEmail is disabled?

No. It is related to that when you disable UsernameIsEmail Umbraco doesnt validate in same username as when you change password.

So case 1:
- disable UsernameIsEmail 
- Create user with name cointaining space
- Click save 
by default it should fail.

Case 2:
- disable UsernameIsEmail 
- setup UserOptions.AllowedUserNameCharacters to allow space
- Create user with name cointaining space
- Click save 
by default it should pass

Case 3:
- disable UsernameIsEmail 
- setup UserOptions.AllowedUserNameCharacters to allow space
- Create user with name cointaining space
-  Click save  
- remove Space from   UserOptions.AllowedUserNameCharacters
- Click save  - here should fail but it will not
- Click change password 
- confirm change of passwor

It will fails as Validation for user name fails. So in case 3 we have the bug from this ticket on Updating user with space when space is not allowed.

**Comment by jerpenol** [comment]
Unfortunately still seeing this in 13.2.2 for Members.

I can add disallowed characters (for example a space or a pipe) to the login field (*_umb_login*) when editing a member, but when I then try to change the password it fails, because the username is not valid (the MemberController.PostSave fails: see the error below).

![image](https://github.com/umbraco/Umbraco-CMS/assets/33028272/cd8d43dd-56e0-46d3-a9ac-98e95c1391bc)

**Comment by AndyButland** [comment]
I'll close this one now and relate to https://github.com/umbraco/Umbraco-CMS/issues/17347 which discusses the same issue, and we've marked as a sprint candidate.
